backend/plugin/parser/mysql/differ.go

```diff
@@ -1,1588 +1,2202 @@
-	createUnsupportedStatement      []string
-	inPlaceDropUnsupportedStatement []string
-	inPlaceAddUnsupportedStatement  []string
-}
-
-func (diff *diffNode) diffUnsupportedStatement(oldUnsupportedStmtList, newUnsupportedStmtList []string) error {
-	// We compare the CREATE TRIGGER/EVENT/FUNCTION/PROCEDURE statements based on strcmp.
-	oldUnsupportedMap, err := buildUnsupportedObjectMap(oldUnsupportedStmtList)
-	if err != nil {
-		return err
-	}
-	newUnsupportedMap, err := buildUnsupportedObjectMap(newUnsupportedStmtList)
-	if err != nil {
-		return err
-	}
-	for tp, objects := range newUnsupportedMap {
-		for newName, newStmt := range objects {
-			if oldStmt, ok := oldUnsupportedMap[tp][newName]; ok {
-				if strings.Compare(oldStmt, newStmt) != 0 {
-					// We should drop the old function and create the new function.
-					// https://dev.mysql.com/doc/refman/8.0/en/drop-procedure.html
-					// https://dev.mysql.com/doc/refman/5.7/en/drop-procedure.html
-					diff.inPlaceDropUnsupportedStatement = append(diff.inPlaceDropUnsupportedStatement, fmt.Sprintf("DROP %s IF EXISTS `%s`;", tp, newName))
-					diff.inPlaceAddUnsupportedStatement = append(diff.inPlaceAddUnsupportedStatement, newStmt)
-				}
-				delete(oldUnsupportedMap[tp], newName)
-				continue
-			}
-			// Now, the input of differ comes from the our mysqldump, mysqldump use ;; to separate the CREATE TRIGGER/FUNCTION/PROCEDURE/EVENT statements;
-			// So we should append DELIMITER statement to the newStmt.
-			delimiterNewStmt := fmt.Sprintf("DELIMITER ;;\n%s\nDELIMITER ;\n", newStmt)
-			diff.createUnsupportedStatement = append(diff.createUnsupportedStatement, delimiterNewStmt)
-		}
-	}
-	// drop remaining TiDB unsupported objects
-	for tp, objects := range oldUnsupportedMap {
-		for name := range objects {
-			diff.dropUnsupportedStatement = append(diff.dropUnsupportedStatement, fmt.Sprintf("DROP %s IF EXISTS `%s`;", tp, name))
-		}
-	}
-
-	return nil
+	dropForeignKeyList        []*foreignKeyDef
+	dropPrimaryKeyList        []*primaryKeyDef
+	createPrimaryKeyList      []*primaryKeyDef
+	dropIndexList             []*indexDef
+	dropViewList              []*viewDef
+	dropTableList             []*tableDef
+	dropCheckConstraintList   []*checkDef
+	addCheckConstraintList    []*checkDef
+	dropIndexConstraintList   []*indexConstraintDef
+	createIndexConstraintList []*indexConstraintDef
+
+	createTableList      []*tableDef
+	alterTableOptionList []*tableOptionDef
+	addColumnList        []*columnDef
+	modifyColumnList     []*columnDef
+	dropColumnList       []*columnDef
+	createTempViewList   []*viewDef
+	createIndexList      []*indexDef
+	addForeignKeyList    []*foreignKeyDef
+	createViewList       []*viewDef
+
+	dropEventList       []*eventDef
+	createEventList     []*eventDef
+	dropTriggerList     []*triggerDef
+	createTriggerList   []*triggerDef
+	dropFunctionList    []*functionDef
+	createFunctionList  []*functionDef
+	dropProcedureList   []*procedureDef
+	createProcedureList []*procedureDef
```

Comment:
As the comments, we should define the member as safe order, which is the deparse order.